tests/integration/test_agentd/test_agentd_multi_server/test_agentd_multi_server.py

```diff
@@ -0,0 +1,213 @@
+
+type: integration
+
+brief: A Wazuh cluster is a group of Wazuh managers that work together to enhance the availability
+       and scalability of the service. These tests will check the agent enrollment in a multi-server
+       environment and how the agent manages the connections to the servers depending on their status.
+
+components:
+    - agentd
+
+targets:
+    - agent
+
+daemons:
+    - wazuh-agentd
+    - wazuh-authd
+    - wazuh-remoted
+
+os_platform:
+    - linux
+    - windows
+
+
+os_version:
+    - Arch Linux
+    - Amazon Linux 2
+    - Amazon Linux 1
+    - CentOS 8
+    - CentOS 7
+    - Debian Buster
+    - Red Hat 8
+    - Ubuntu Focal
+    - Ubuntu Bionic
+
+references:
+    - https://documentation.wazuh.com/current/user-manual/registering/index.html
+
+tags:
+    - enrollment
+'''
+import pytest
+from pathlib import Path
+import sys
+
+from wazuh_testing.constants.paths.logs import WAZUH_LOG_PATH
+from wazuh_testing.constants.platforms import WINDOWS
+from wazuh_testing.modules.agentd.configuration import AGENTD_DEBUG, AGENTD_WINDOWS_DEBUG, AGENTD_TIMEOUT
+from wazuh_testing.modules.agentd.patterns import * 
+from wazuh_testing.tools.monitors.file_monitor import FileMonitor
+from wazuh_testing.tools.simulators.authd_simulator import AuthdSimulator
+from wazuh_testing.utils import callbacks
+from wazuh_testing.utils.client_keys import add_client_keys_entry
+from wazuh_testing.utils.configuration import get_test_cases_data, load_configuration_template
+
+from . import CONFIGS_PATH, TEST_CASES_PATH
+
+# Marks
+pytestmark = pytest.mark.tier(level=0)
+
+# Configuration and cases data.
+configs_path = Path(CONFIGS_PATH, 'wazuh_conf.yaml')
+cases_path = Path(TEST_CASES_PATH, 'cases_reconnection_protocol.yaml')
+
+# Test configurations.
+config_parameters, test_metadata, test_cases_ids = get_test_cases_data(cases_path)
+test_configuration = load_configuration_template(configs_path, config_parameters, test_metadata)
+
+if sys.platform == WINDOWS:
+    local_internal_options = {AGENTD_WINDOWS_DEBUG: '2'}
+else:
+    local_internal_options = {AGENTD_DEBUG: '2'}
+local_internal_options.update({AGENTD_TIMEOUT: '5'})
+
+daemons_handler_configuration = {'all_daemons': True}
+
+# Tests
+"""
+How does this test work:
+    - PROTOCOL: tcp/udp
+    - DELETE_KEYS_FILE: whatever start with an empty client.keys file or not
+    - SIMULATOR MODES: for each simulator will define a mode
+    - AUTHD_PREV_MODE: represents the previous state of the authd daemon,
+        - ACCEPT means it already has keys, REJECT means no keys
+    - AUTHD: mode for the authd simulator
+    - LOG_MONITOR_STR: (list of lists) Expected string to be monitored for each simulator
+"""
+@pytest.mark.parametrize('test_configuration, test_metadata', zip(test_configuration, test_metadata), ids=test_cases_ids)
+def test_agentd_multi_server(test_configuration, test_metadata, set_wazuh_configuration, configure_local_internal_options, truncate_monitored_files, 
```

Comment:
Not all fixtures are in the description. Check all tests.

Reply:
Addressed with https://github.com/wazuh/wazuh/pull/18353/commits/5a1b7778d254f00b85c507ace508492c18df7d0c